packages/react-core/src/components/EmptyState/EmptyState.tsx

```diff
@@ -10,37 +13,79 @@ export enum EmptyStateVariant {
   full = 'full'
 }
 
+export type EmptyStateStatus = 'danger' | 'warning' | 'success' | 'info' | 'custom';
+export type EmptyStateVariantType = 'xs' | 'sm' | 'lg' | 'xl' | 'full';
+
 export interface EmptyStateProps extends React.HTMLProps<HTMLDivElement> {
   /** Additional classes added to the empty state */
   className?: string;
   /** Content rendered inside the empty state */
-  children: React.ReactNode;
+  children?: React.ReactNode;
   /** Modifies empty state max-width and sizes of icon, title and body */
-  variant?: 'xs' | 'sm' | 'lg' | 'xl' | 'full';
+  variant?: EmptyStateVariantType;
   /** Cause component to consume the available height of its container */
   isFullHeight?: boolean;
+  /** Status of the empty state, will set a default status icon and color. Icon can be overwritten using the icon prop */
+  status?: EmptyStateStatus;
+  /** Additional class names to apply to the empty state header */
+  headerClassName?: string;
+  /** Additional props passed to the empty state header container */
+  headerProps?: React.HTMLProps<HTMLDivElement>;
+  /** Additional classes added to the title inside empty state header */
+  titleClassName?: string;
+  /** Text of the title inside empty state header, will be wrapped in headingLevel */
+  titleText: React.ReactNode;
+  /** Empty state icon element to be rendered. Can also be a spinner component */
+  icon?: React.ComponentType<any>;
+  /** Additional props passed to the icon element */
+  iconProps?: EmptyStateIconProps;
+  /** The heading level to use, default is h1 */
+  headingLevel?: EmptyStateHeadingLevel;
 }
 
 export const EmptyState: React.FunctionComponent<EmptyStateProps> = ({
   children,
   className,
-  variant = EmptyStateVariant.full,
+  variant = 'full',
   isFullHeight,
+  status,
+  icon: customIcon,
+  titleText,
+  titleClassName,
+  headerClassName,
+  headingLevel,
+  headerProps,
   ...props
-}: EmptyStateProps) => (
-  <div
-    className={css(
-      styles.emptyState,
-      variant === 'xs' && styles.modifiers.xs,
-      variant === 'sm' && styles.modifiers.sm,
-      variant === 'lg' && styles.modifiers.lg,
-      variant === 'xl' && styles.modifiers.xl,
-      isFullHeight && styles.modifiers.fullHeight,
-      className
-    )}
-    {...props}
-  >
-    <div className={css(styles.emptyStateContent)}>{children}</div>
-  </div>
-);
+}: EmptyStateProps) => {
+  const statusIcon = status && statusIcons[status];
+  const icon = customIcon || statusIcon;
```

Comment:
One decision point: if we have the status prop set a default icon should the `icon` prop override it, or the other way around?

I think as a consumer I would prefer the former so that I can have the status apply custom colors to my passed icon, so that's what I went with, but I'm not dead set.

Reply:
Agreed. I only have one hesitation with that, but it's something that can be covered in a11y docs (passing an icon that doesn't visually represent the status)